cirq-core/cirq/experiments/qubit_characterizations.py

```diff
@@ -36,7 +37,37 @@
     import cirq
 
 
-@dataclasses.dataclass
+def _canonize_clifford_sequences(
+    sequences: list[list[ops.SingleQubitCliffordGate]],
+) -> list[tuple[ops.SingleQubitCliffordGate]]:
+    return [(_reduce_gate_seq(seq),) for seq in sequences]
+
+
+@attrs.frozen
+class _CliffordGateSequence:
+    """Wrap around a list of sequences of clifford gates.
+
+    This class wraps around a list of sequences of clifford gates and re-exposes them as
+    a list of tuple where each tuple contains a single clifford gates.
+    """
+
+    gate_sequence: list[list[ops.SingleQubitCliffordGate]]
+
+    @functools.cached_property
+    def _reduced_gate_sequence(self) -> list[tuple[ops.SingleQubitCliffordGate]]:
+        return _canonize_clifford_sequences(self.gate_sequence)
+
+    def __iter__(self) -> Iterator[tuple[ops.SingleQubitCliffordGate]]:
+        yield from self._reduced_gate_sequence
+
+    def __getitem__(self, idx) -> tuple[ops.SingleQubitCliffordGate]:
+        return self._reduced_gate_sequence[idx]
+
+    def __len__(self) -> int:
+        return len(self._reduced_gate_sequence)
+
+
+@attrs.frozen
 class Cliffords:
     """The single-qubit Clifford group, decomposed into elementary gates.
 
```

Comment:
Please add a short note to docstring the arguments are converted to a reduced gate sequence.